### PR TITLE
[test] Add validation test on range pickers

### PR DESCRIPTION
**Pull request by alexfauquette**

### CODE REVIEW

File: packages/x-date-pickers-pro/src/DateRangePicker/DateRangePicker.test.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { DateRangePicker } from '@mui/x-date-pickers-pro/DateRangePicker';
 import { fireEvent, screen } from '@mui/monorepo/test/utils/createRenderer';
 import { expect } from 'chai';
 import { createPickerRenderer, stubMatchMedia, wrapPickerMount } from 'test/utils/pickers-utils';
+import describeValidation from '../tests/describeValidation';
```
Comment (flaviendelangle) [inline packages/x-date-pickers-pro/src/DateRangePicker/DateRangePicker.test.tsx:8]: Should it be an absolute path like on the community package ?

File: packages/x-date-pickers-pro/src/tests/describeValidation.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+const testsToRun = {
+  // day: testDayViewValidation,
+  textField: testTextFieldValidation,
+};
+
+interface ConformanceOptions {
+  render: any;
+  clock: any;
+  after?: () => void;
```
Comment (flaviendelangle) [inline packages/x-date-pickers-pro/src/tests/describeValidation.tsx:39]: I don't seea ny usage of this callback, including in the community version, what would be the use case for it ?

Comment (alexfauquette) [reply packages/x-date-pickers-pro/src/tests/describeValidation.tsx:39]: No idea, I assume if the core team added such option it was for good reason but Yes for now we don't need it

Comment (flaviendelangle) [reply packages/x-date-pickers-pro/src/tests/describeValidation.tsx:39]: Oh you just took it from `describeConformance` ? OK to keep the same options for now even  if not used

File: packages/x-date-pickers-pro/src/tests/describeValidation.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+/**
+ * Tests various aspects of picker validation
+ * components.
+ * @param {React.ReactElement} minimalElement - the picker to test
+ * @param {() => ConformanceOptions} getOptions
+ */
+export default function describeValidation(minimalElement, getOptions: () => ConformanceOptions) {
+  describe('RangePickers validation API', () => {
```
Comment (flaviendelangle) [inline packages/x-date-pickers-pro/src/tests/describeValidation.tsx:56]:
```suggestion
  describe('Range pickers validation API', () => {
```

**Review by flaviendelangle — APPROVED** [review]

**Pull request merged**